### PR TITLE
bug fix > is.palindrome

### PR DESCRIPTION
**Pull request by ironmaniiith**
Refer #176

### CODE REVIEW

File: is.js [hunk 1]
```diff
@@ -399,7 +399,17 @@
 
     // is a given string palindrome?
     is.palindrome = function(str) {
-        return is.string(str) && str === str.split('').reverse().join('');
+        if (is.not.string(str)) {
+            return false;
+        }
+        var str = str.replace(/[^a-zA-Z0-9]+/gi, '').toLowerCase();
```
Comment (jdalton) [inline is.js]: No need for the `i` flag since you've included the `a-z` and `A-Z` variants.

File: is.js [hunk 2]
```diff
@@ -399,7 +399,17 @@
...
+        var len = str.length - 1;
```
Comment (jdalton) [inline is.js]: Can you store the length with it divided by `2` in `len`.

Comment (ironmaniiith) [inline is.js]: May I know the reason?
I mean is it for maintaining the proper code or something else.?

**Comment by jdalton** [comment]
🙌

**Pull request merged**